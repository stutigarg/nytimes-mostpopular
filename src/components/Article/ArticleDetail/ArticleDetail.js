import React from 'react';
import Loader from "../../../util/Loader";

const ArticleDetail = ({article}) => {
  return ( article ?
      <iframe src={article.url}
              allowFullScreen
              title={article.title}
              frameBorder="0"
              position="relative" height="100%" width="100%"/>
              : <div><Loader/></div>
  );
}

export default ArticleDetail;
