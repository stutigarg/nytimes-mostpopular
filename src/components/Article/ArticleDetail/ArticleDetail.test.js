import React from 'react';
import {configure, shallow} from 'enzyme';
import ArticleDetail from './ArticleDetail';
import Adapter from 'enzyme-adapter-react-16';

configure({adapter: new Adapter()});

const article = {
  url:'http://google.com/search-there',
  title: 'A really interesting article.',
};

describe('<ArticleDetail />', () => {
  test('renders article details', () => {
    const wrapper = shallow(<ArticleDetail article={article}/>);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('iframe').length).toBe(1);
    expect(wrapper.find('Loader').length).toBe(0);
  });
});

describe('<ArticleDetail />', () => {
  test('spins loader without article', () => {
    const wrapper = shallow(<ArticleDetail />);
    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find('Loader').length).toBe(1);
    expect(wrapper.find('iframe').length).toBe(0);
  });
});
