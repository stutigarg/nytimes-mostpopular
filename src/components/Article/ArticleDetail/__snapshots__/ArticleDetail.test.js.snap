// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticleDetail /> renders article details 1`] = `ShallowWrapper {}`;

exports[`<ArticleDetail /> spins loader without article 1`] = `ShallowWrapper {}`;
